odoo/addons/base/models/res_country.py

```diff
@@ -106,7 +111,10 @@ def create(self, vals_list):
     def write(self, vals):
         if vals.get('code'):
             vals['code'] = vals['code'].upper()
-        return super(Country, self).write(vals)
+        res = super(Country, self).write(vals)
+        self.flush()
```

Comment:
why `flush()`?

Reply:
To make sure that the `write()` is effective before the next read that will re-populate the cache.
The same approach is used in `res.lang`'s `write`, which is what led me into thinking this was needed.

Reply:
Hmm, the next `read()` during the same request would read it from the env cache where it was written (flushed or not), so it would always see the new value.

And if that `read()` happened in another request/transaction, it wouldn't see the new value until the transaction that wrote it was fully committed. And that happens independently of any `flush()`.

Generally, flushing to the database is the job of the framework and it should be transparent. You shouldn't need to do it yourself unless you're bypassing the normal mechanisms by e.g. searching directly with `cr.execute("SELECT ...")`. In that case you need to (partially) flush because otherwise you may fail to include data that hasn't been flushed to the database yet.

So here I don't think it's needed at all.

~~

The `flush()` in `res.lang` was introduced during the ORM refactoring for 13.0 (9920f20e4c7753bc17bea71dea3a90f7de687196), and it would have deserved a comment, because the rationale is really unclear. It may not even be needed.

Also, the `clear_caches()` in `res.lang` should have been more specific ;)

Reply:
Thanks a lot for these explanations.

Just to make sure: this is also true cross-workers: the `clear_caches` signaling is done within the same SQL transaction as the actual changes - and it therefore appears as an atomic operation to other workers, right ?